### PR TITLE
Add MCMC fitting for GPyTorch

### PR DESCRIPTION
**Pull request by kiranvad**
This PR adds following additions to the `PyTorchExtrapolator.py` module:
1. Along with the gradient descent based optimization of marginal likelihood (in `mll`), a fully Bayesian approach is included under `mcmc` method of the module to fit the GP Models. 
2. It also returns gradient of entropy function using PyTorch's `autograd` module that is useful in visualizing phase boundaries and transitions following [this paper](https://chemrxiv.org/engage/chemrxiv/article-details/67930f56fa469535b99a0dd1). 
3. GP model kernel now uses (smooth) Matern Kernel with automatic relevance determination.

### CODE REVIEW

**Comment by kiranvad** [comment]
ToDos:
- [ ] Add test functions for the `PyTorchExtrapolator` class

**Comment by kiranvad** [comment]
addressed in #104 and no longer required
